crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -269,6 +290,39 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
     }
 }
 
+#[cfg(target_family = "unix")]
+fn unix_read_paths_from_stdin() -> Result<Vec<PathBuf>, proc_exit::Exit> {
+    use std::ffi::OsString;
+    use std::os::unix::ffi::OsStringExt;
+
+    let mut buf = Vec::new();
+    io::stdin()
+        .read_to_end(&mut buf)
+        .with_code(proc_exit::sysexits::IO_ERR)?;
+
+    let (mut paths, rest) = buf.into_iter().fold(
+        (vec![], vec![]),
+        |(mut paths, mut cur_path): (Vec<PathBuf>, Vec<u8>), byte| {
+            if byte == 0 {
+                paths.push(PathBuf::from(OsString::from_vec(cur_path)));
+                (paths, vec![])
+            } else {
+                cur_path.push(byte);
+                (paths, cur_path)
+            }
+        },
+    );
+    if !rest.is_empty() {
+        paths.push(PathBuf::from(OsString::from_vec(rest)));
+    }
+    Ok(paths)
+}
+
+#[cfg(not(target_family = "unix"))]
+fn unix_read_paths_from_stdin() -> Result<Vec<PathBuf>, proc_exit::Exit> {
+    proc_exit::sysexits::CONFIG_ERR.ok()?;
+}
+
```

Comment:
Options
- We keep this platform-specific
- We use `os_str_bytes`
- We wait on rust-lang/rust#109698
- We guiltily act as if rust-lang/rust#109698 is reality and transmute

Reply:
- We punt on null terminated paths for now

Reply:
I don't like waiting. If we make it platform-specific now, some good brave soul can make it works on Windows once they need it.

`os_str_bytes` seems fine with me if you're OK with an extra dependency.

Reply:
`os_str_bytes` is already a dependency through clap (until we upgrade; it was dropped recently)

Reply:
So it seems like it's not wanted?

I don't really have much time to spend on this and I don't care about Windows, so I would rather leave as is. It doesn't make anything worse for Windows users, AFAICT.

Reply:
https://docs.rs/os_str_bytes/latest/os_str_bytes/struct.EncodingError.html

> [Result::unwrap](https://doc.rust-lang.org/nightly/core/result/enum.Result.html#method.unwrap) should almost always be called on results containing this error. It should be known whether or not byte sequences are properly encoded for the platform, since [the module-level documentation](https://docs.rs/os_str_bytes/latest/os_str_bytes/index.html#encoding) discourages using encoded bytes in interchange. Results are returned primarily to make panicking behavior explicit.

This sounds like it is not supposed to be used on user provided data (the whole crate)? When I tried to use it anyway, it complained about having to set some env variable.

Anyway. I'm out of time and I only go out of my way to support commercial software when paid to do it. Its fine if you don't want to land it.